### PR TITLE
Addition of cnv data to oncoprint

### PR DESCRIPTION
**Pull request by cbethell**
### Purpose/implementation

This PR incorporates the cnv data into the data represented on the oncoprint produced in PR #176.

It builds onto the `01-plot-oncoprint.R` and `run-oncoprint.sh` scripts in the oncoprint-landscape module. 

### Issue

This PR addresses issue #6 on producing an oncoprint that displays the landscape of genetic lesions across PBTA.

### Directions for reviewers

- In this PR, I changed the option to `removeNonMutated` from `FALSE` to `TRUE` so that the `Tumor_Sample_Barcode`s would be readable on the plot. Should I set an optparse option to toggle this off and on? Should I leave it as is or set it back to `FALSE`?
- I determined the threshold for distinguishing amplifications, homozygous deletions, and hemizygous deletions using the CNVkit documentation [here](https://cnvkit.readthedocs.io/en/stable/calling.html). However, their threshold was too low to display amplifications in the selected genes, so I made the thresholds optparse options and used a cutoff of 0.2 instead of 0.5 to produce the current plot. Should I change this threshold? Perhaps this question may be better answered once we receive the consensus calls, as mentioned on the previous PR #176. 
- This script has been getting a bit lengthy, should I now break it up into multiple scripts? If so, what would be your recommendation for the most logical way to do so? 

### Results

See the current plot below:
maf_oncoprint

#### Docker and continuous integration
- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### PR Checklist
* [x] Run a linter
* [x] Set the seed- Not needed.
* [x] Comments and/or documentation up to date 
* [x] Double check your paths
* [x] Spell check any Rmd file or md file
* [x] Restart R and run all notebooks fresh and save

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
Although, this probably will be broader than *just* gene symbols.

File: analyses/oncoprint-landscape/01-plot-oncoprint.R [hunk 1]
```diff
@@ -116,11 +129,57 @@ metadata <- metadata %>%
...
+
+  # Create a dataframe with `Tumor_Sample_Barcode` and gene information from maf
+  # dataframe
+  select_maf_df <- maf_df %>%
+    dplyr::select(Tumor_Sample_Barcode, Hugo_Symbol)
+
+  # Add gene information to the cnv dataframe
+  cnv_df <- cnv_file %>%
+    dplyr::inner_join(select_maf_df, by = c("ID" = "Tumor_Sample_Barcode")) %>%
+    dplyr::distinct()
```
[the remark this answers is not in the document]
Comment (jharenza) [reply analyses/oncoprint-landscape/01-plot-oncoprint.R]: @cbethell - I have some code in which I created the focal copy number for oncoprints from seg file [here](https://github.com/marislab/pptc-pdx-copy-number-and-SVs/blob/master/R/focal-CN-revision.R) if you want to take a look. Sorry it is a bit messy (and still has old GISTIC code - did not have time to clean that up yet, but just posted so you can see it). Note, as @jaclyn-taroni said, that these old data were hg19 and the PBTA data are hg38. The seg file for input to this is [here](https://ndownloader.figshare.com/files/14414165) and the output is focal copy number lesions used for oncoprints [here](https://ndownloader.figshare.com/files/16603340). Note, this was also done using array data, not NGS, so we may have to figure out thresholds for cutoffs for accurate assessment of focal CN based on LRR (may advise starting with ATRT's and SMARCB1 deletions and we may not be able to do homozygous/hemizygous, but rather just deletions/amplifications - not sure - haven't dug into the data too much). I ended up doing a lot of manual inspection for driver genes, hence the re-coding in the script. Maybe this should be its own issue?

[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply analyses/oncoprint-landscape/01-plot-oncoprint.R]: Thank you for sharing that @jharenza

**Comment by cbethell** [comment]
I am going to close this PR. The plan is to open a separate PR related specifically to the CN matrix described in @jharenza's comment [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/182#discussion_r340587968).

**Pull request closed without merging**